aten/src/ATen/native/cpu/ScatterGatherKernel.cpp

```diff
@@ -816,8 +815,7 @@ void cpu_scatter_reduce_expanded_index(const Tensor& self, const Tensor& _index,
 }
 
 template <typename scalar_t>
-void cpu_gather_expanded_index_kernel(const Tensor& result, const Tensor& _index, const Tensor& self) {
-  Tensor index = _index.to(ScalarType::Long);
+void cpu_gather_expanded_index_kernel(const Tensor& result, const Tensor& index, const Tensor& self) {
```

Comment:
I think you still need it here? A call from top level `torch.gather` can bring you here with int32 indices (don't know if there are tests for this, probably should be added if not) https://github.com/pytorch/pytorch/blob/eb19f5abab4a8295421a08a2417d1a51bff51521/aten/src/ATen/native/TensorAdvancedIndexing.cpp#L2084

Reply:
I added it back. Is there an easy way to trigger that code path?

Reply:
I'm not sure, probably same trigger that allows us to take vectorized path on the cuda side? `gather` on dim 0, with index expanded in dim 1, but you could make sure looking at the function they use to select it (`can_use_gather_expanded` or something like that)